Source/WebCore/platform/graphics/gstreamer/mse/WebKitMediaSourceGStreamer.cpp

```diff
@@ -70,7 +70,6 @@ struct WebKitMediaSrcPrivate {
     {
         ASSERT(isMainThread());
         Stream* stream = streams.get(name);
-        ASSERT(stream);
```

Comment:
In what situations are WebKitMediaSrc functions called with an inexisting stream?

Reply:
When `webKitMediaSrcFlush` is called after `webKitMediaSrcTearDownStream`, which I believe shouldn't normally happens, but it does in this test. The details that make this to happen in the test are:

1. When the page loads, the video element is hidden. This will trigger the `!isVisible` branch in `MediaPlayerPrivateGStreamer::setVisibleInViewport`. In that branch, the first state change to PAUSED fails because the playbin URL isn't set, leaving the pipeline in READY with PAUSED pending, but the webkitmediasrc element is left already in PAUSED.
2. When the video is visible, it's also seeked (from the test). But the pipeline is still in READY with PAUSED pending.
3. When the video is hidden again, `MediaPlayerPrivateGStreamer::setVisibleInViewport` will keep `m_invisiblePlayerState = READY`, because that's state of the pipeline at that moment - it doesn't transition to PAUSED even after the URL is set and `gst_element_set_state` is called again.
4. When the video is visible again, this time `MediaPlayerPrivateGStreamer::setVisibleInViewport` changes the pipeline state to READY (it uses `m_invisiblePlayerState` from the previous step). This changes webkitmediasrc to READY as well, and that calls `webKitMediaSrcTearDownStream` which clears the streams. Then, a new seek is attempted (from the test), that will trickle down to `webKitMediaSrcFlush`, but this time webkitmediasrc has no streams and crashes dereferencing the stream pointer returned from `streamByName`.

Reply:
No need to change that part anymore.